### ISSUE TITLE
typescript: don't warn on computed properties [phab:D1667]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2091]
_Created by **evmar** at 2017-05-23 18:15:08_: 

Computed properties are runtime-determined attributes;
we can't really cross-reference them, but we should verify
that we cross-reference the expressions contained within
the computed property.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-05-23 18:17:41_:

It seems odd to generate a ref to something that will have no properties. If there isn't even a node kind to point to, the UI won't be able to even enumerate the reference. 

The structure looks fine, but I wonder if it's worth emitting.

**Comment by kythe** [comment]
_**fromberger** wrote at 2017-05-23 18:17:41_:

"Computed property" is a fancy way of saying "macro" apparently? :)

**Comment by kythe** [comment]
_**evmar** commented inline (context lost during migration) at 2017-05-23 18:24:35_:

The referred-to thing here has properties (it's a builtin object), but I didn't think this test should be asserting on the properties of that thing.  I'm just verifying that the indexing "looks in" to the expression inside the [].

**Comment by kythe** [comment]
_**evmar** wrote at 2017-05-23 18:24:35_:

Computed property is this weird syntax for object literals, where

```
var x = {
  foo: 3,
};
```

creates an object with the static property named 'foo', while

```
var x = {
  [foo]: 3,
}
```

creates an object with a property named whatever the *value* of the foo expression is.

It can be used to create maps, e.g.

```
var map = {
  [constants.FOO]: 3,
  [constants.BAR]: 4
};
```

**Comment by kythe** [comment]
_**evmar** wrote at 2017-05-23 18:25:34_:

also index into function body

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-05-23 20:05:48_:

I just wonder if the test shouldn't at least verify that the targets of those references //could// be the correct object. I agree it's not worth checking all the properties, but maybe the kind and/or the signature? 

Right now, with a blank EVar, it could literally be anything (and indeed, both could refer to the same thing).

**Comment by kythe** [comment]
_**evmar** commented inline (context lost during migration) at 2017-05-24 23:07:51_:

OK, I added a signature here.

**Comment by kythe** [comment]
_**evmar** wrote at 2017-05-24 23:08:14_:

more vnames

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2017-05-24 23:10:01_

**Comment by kythe** [comment]
_**fromberger** closed via commit 5b0454e8d1fee65fd5d0b08287a3ca8469dfd44d at 2017-05-24 23:10:35_
